src/Heading.js

```diff
@@ -19,5 +28,40 @@ Heading.propTypes = {
   ...COMMON.propTypes,
   ...TYPOGRAPHY.propTypes
 }
+//
+
+const HeadingInternal = styled.h2`
+  font-weight: ${get('fontWeights.bold')};
+  margin: 0;
+  ${TYPOGRAPHY} ${COMMON};
+`
+
+HeadingInternal.defaultProps = {
+  theme
+}
+
+HeadingInternal.propTypes = {
+  theme: PropTypes.object,
+  ...COMMON.propTypes,
+  ...TYPOGRAPHY.propTypes
+}
+
+function withHeading(level) {
+  const WithHeading = props => (
+    <HeadingInternal
+      as={level}
+      fontSize={[`mobileHeadingSizes.${level}`, `mobileHeadingSizes.${level}`, `headingSizes.${level}`]}
+      {...props}
+    />
+  )
+  WithHeading.propTypes = HeadingInternal.propTypes
+  WithHeading.defaultProps = HeadingInternal.defaultProps
+  return WithHeading
+}
 
-export default Heading
+export const H1 = withHeading('h1')
+export const H2 = withHeading('h2')
+export const H3 = withHeading('h3')
+export const H4 = withHeading('h4')
+export const H5 = withHeading('h5')
+export const H6 = withHeading('h6')
```

Comment:
If `Heading` is being deprecated then it should probably still be exported as the default export so that the imports don't break. The TS module will need to be updated accordingly.

Reply:
Does it matter though since we're exporting everything as named exports in `index.js`? We aren't importing Heading in any other component files so we shouldn't be relying on the default export anywhere

Reply:
The previous module definition for `src/Heading` was:

```typescript
declare module '@primer/components/src/Heading' {
  import {Heading} from '@primer/components'
  export default Heading
}
```

so if we change `Heading.js` so that `Heading` isn't being exported as default, that's a breaking change for this TypeScript module.

Reply:
Also #789 will make individual modules requireable, so we will need to stick to an export contract for those.